Phoenix/PHSpace.m

```diff
@@ -223,8 +244,23 @@ - (void)removeWindows:(NSArray<PHWindow *> *)windows {
 }
 
 - (void)moveWindows:(NSArray<PHWindow *> *)windows {
-    CGSMoveWindowsToManagedSpace(
-        CGSMainConnectionID(), (__bridge CFArrayRef)[self identifiersForWindows:windows], self.identifier);
+    CGSConnectionID cid = CGSMainConnectionID();
+    if ([NSProcessInfo canMoveWindowsToManagedSpace]) {
+        CGSMoveWindowsToManagedSpace(cid, (__bridge CFArrayRef)[self identifiersForWindows:windows], self.identifier);
+        return;
+    }
+
+    NSArray<NSNumber *> *ids = [self identifiersForWindows:windows];
+    NSUInteger numIds = [ids count];
+    NSMutableData *contiguousIds = [[NSMutableData alloc] initWithCapacity:numIds * sizeof(CGWindowID)];
+    for (id item in ids) {
+        CGWindowID value = [item unsignedIntValue];
+        [contiguousIds appendBytes:&value length:sizeof(CGWindowID)];
+    }
+
+    CGSSpaceSetCompatID(cid, self.identifier, MoveWindowsCompatId);
+    CGSSetWindowListWorkspace(cid, (CGWindowID *)[contiguousIds bytes], numIds, MoveWindowsCompatId);
+    CGSSpaceSetCompatID(cid, self.identifier, 0x0);
```

Comment:
What is the `0x0` for?

Reply:
I copied it from the yabai and Amethyst fixes - I assume it's to unset the compat ID, but there's no clear documentation.